File: backend/app/schemas/trading.py
from pydantic import BaseModel, Field
from typing import Optional, List, Dict, Any
from datetime import datetime


# 订单相关
class OrderCreate(BaseModel):
    symbol: str = Field(..., description="交易对，如 BTC-USDT")
    side: str = Field(..., description="买卖方向: buy/sell")
    order_type: str = Field(..., description="订单类型: limit/market")
    quantity: float = Field(..., gt=0, description="数量")
    price: Optional[float] = Field(None, gt=0, description="限价单价格")
    api_key_id: Optional[int] = Field(None, description="使用的API密钥ID")


class OrderResponse(BaseModel):
    id: int
    exchange_order_id: Optional[str]
    symbol: str
    side: str
    order_type: str
    price: Optional[float]
    quantity: float
    filled_quantity: float
    total_amount: Optional[float]
    filled_amount: float
    status: str
    created_at: datetime
    updated_at: Optional[datetime]
    filled_at: Optional[datetime]

    class Config:
        from_attributes = True


class OrderCancelResponse(BaseModel):
    success: bool
    message: str
    order_id: Optional[int] = None


# 成交记录相关
class TradeRecordResponse(BaseModel):
    id: int
    symbol: str
    side: str
    price: float
    quantity: float
    amount: float
    fee: float
    fee_currency: Optional[str]
    pnl: Optional[float]
    pnl_percent: Optional[float]
    traded_at: datetime

    class Config:
        from_attributes = True


# 持仓相关
class PositionResponse(BaseModel):
    symbol: str
    quantity: float
    avg_price: float
    current_price: float
    unrealized_pnl: float
    unrealized_pnl_percent: float
    market_value: float


# 余额相关
class BalanceResponse(BaseModel):
    currency: str
    available: float
    frozen: float
    total: float
    usd_value: Optional[float] = None


# 分页响应
class PaginatedResponse(BaseModel):
    items: List[Any]
    total: int
    page: int
    page_size: int
    total_pages: int


# K线数据
class KlineData(BaseModel):
    timestamp: int
    open: float
    high: float
    low: float
    close: float
    volume: float


# 深度数据
class DepthData(BaseModel):
    asks: List[List[float]]  # 卖盘 [price, quantity]
    bids: List[List[float]]  # 买盘 [price, quantity]
    timestamp: int


# 行情数据
class TickerData(BaseModel):
    symbol: str
    last_price: float
    price_change_percent: float
    high_24h: float
    low_24h: float
    volume_24h: float
    turnover_24h: float

File: src/api/trading.ts
import apiClient from './request'

export interface OrderCreate {
  symbol: string
  side: 'buy' | 'sell'
  order_type: 'limit' | 'market'
  quantity: number
  price?: number
  api_key_id?: number
}

export interface OrderResponse {
  id: number
  exchange_order_id?: string
  symbol: string
  side: string
  order_type: string
  price?: number
  quantity: number
  filled_quantity: number
  total_amount?: number
  filled_amount: number
  status: string
  created_at: string
  updated_at?: string
  filled_at?: string
}

export interface TradeRecord {
  id: string
  symbol: string
  side: string
  price: number
  quantity: number
  amount: number
  fee: number
  fee_currency: string
  pnl?: number
  pnl_percent?: number
  traded_at: string
}

export interface Position {
  symbol: string
  quantity: number
  avg_price: number
  current_price: number
  unrealized_pnl: number
  unrealized_pnl_percent: number
  market_value: number
}

export interface Balance {
  currency: string
  available: number
  frozen: number
  total: number
  usd_value?: number
}

export interface PaginatedResponse<T> {
  items: T[]
  total: number
  page: number
  page_size: number
  total_pages: number
}

// 获取账户余额
export function getBalance(apiKeyId?: number): Promise<Balance[]> {
  return apiClient.get('/trading/balance', {
    params: { api_key_id: apiKeyId }
  })
}

// 获取持仓
export function getPositions(instType = 'SPOT', apiKeyId?: number): Promise<Position[]> {
  return apiClient.get('/trading/positions', {
    params: { inst_type: instType, api_key_id: apiKeyId }
  })
}

// 下单
export function placeOrder(data: OrderCreate): Promise<OrderResponse> {
  return apiClient.post('/trading/orders', data)
}

// 撤单
export function cancelOrder(orderId: number, apiKeyId?: number): Promise<{ success: boolean; message: string; order_id?: number }> {
  return apiClient.delete(`/trading/orders/${orderId}`, {
    params: { api_key_id: apiKeyId }
  })
}

// 获取订单列表（分页）
export function getOrders(params: {
  page?: number
  page_size?: number
  symbol?: string
  status?: string
  side?: string
}): Promise<PaginatedResponse<OrderResponse>> {
  return apiClient.get('/trading/orders', { params })
}

// 获取订单详情
export function getOrderDetail(orderId: number, apiKeyId?: number): Promise<OrderResponse> {
  return apiClient.get(`/trading/orders/${orderId}`, {
    params: { api_key_id: apiKeyId }
  })
}

// 获取成交记录（分页）
export function getTrades(params: {
  page?: number
  page_size?: number
  symbol?: string
  side?: string
  start_date?: string
  end_date?: string
  api_key_id?: number
}): Promise<PaginatedResponse<TradeRecord>> {
  return apiClient.get('/trading/trades', { params })
}

// 获取历史订单（交易所）
export function getOrderHistory(params: {
  page?: number
  page_size?: number
  inst_type?: string
  state?: string
  api_key_id?: number
}): Promise<PaginatedResponse<any>> {
  return apiClient.get('/trading/orders/history', { params })
}
